include/ensmallen_bits/adam/padam_update.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file padam_update.hpp
+ * @author Marcus Edel
+ *
+ * Implementation of Partially adaptive momentum estimation method (Padam).
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADAM_PADAM_UPDATE_HPP
+#define ENSMALLEN_ADAM_PADAM_UPDATE_HPP
+
+namespace ens {
+
+/**
+ * Partially adaptive momentum estimation method (Padam),
+ * adopts historical gradient information to automatically adjust the
+ * learning rate.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{
+ *   title   = {Closing the Generalization Gap of Adaptive Gradient Methods in
+ *              Training Deep Neural Networks},
+ *   author  = {{Chen}, J. and {Gu}, Q.},
+ *   journal = {ArXiv e-prints},
+ *   url     = {https://arxiv.org/abs/1806.06763}
+ *   year    = {2018}
+ * }
+ * @endcode
+ */
+class PadamUpdate
+{
+ public:
+  /**
+   * Construct the Padam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
```

Comment:
Would it be more accurate to say something like "Epsilon is the minimum allowed gradient"?  My understanding of epsilon is that it's necessary to prevent dividing by very small values in the case where v_t is very small.  (Perhaps that might be a documentation update to fix in the AdamUpdate and other similar optimizers---not sure?)